### ISSUE TITLE
Retrained model is not performing as claimed

### ISSUE THREAD

**Issue by dattatrayshinde** [Neerajj9/CSRNet-keras#1]
First of all I would like to appreciate the efforts to make this possible in keras but when I retrained the model on ShanghaiTech Dataset it is not performing well as compared with trained model you have provided. Are the hyper-parameters provided are right? What changes need to be done to get optimum results?

**Comment by Puetz** [comment]
Hi,
I also tried retraining the model using the ShanghaiTech dataset but got really bad results. The data preprocessing seemed to work. I plotted a few images with their corresponding ground truth and they looked just fine.
I tried different learning rates (1e-9 - 1e-5), used 500-700 steps per epoch and 100-200 epochs on each run. I also plotted the loss using a callback and none of the graphs converged to 0. They were all really zig-zagged.
Does anyone know what's causing this? I didn't change anything but the learning rate.

**Comment by aditya-malte** [comment]
Hi Puetz,
As would be visible from the code, our model is exactly as per the paper proposed by the author, even the hyperparameters are the same. Please check again with how you've run the code if there seems to be an anomaly.
Thanks

Regards,
Aditya Malte

**Comment by Puetz** [comment]
Hi,
I cloned the repo again, redid the preprocessing which seemed completely fine and started a new training job this morning. I didn't change anything at the model. I only added one cell for plotting the loss.  
My result with 200 epochs and your default values looks like this:  

![image](https://user-images.githubusercontent.com/34755160/55402421-be9b3500-5553-11e9-8e4a-fb7d04008a21.png)

I uploaded the two notebooks to this [repository](https://github.com/Puetz/csrnet-keras-issue/blob/master/Model.ipynb), so that you can have a look at them.
My package version are 

`Keras==2.2.4
tensorflow-gpu==1.13.1
scipy==1.2.1
numpy==1.16.2
Pillow==5.4.1
opencv-python==4.0.0.21`.

I tried using your versions but CUDA 9.0 did not work on my GPU. Do you think my bad results are related to the specific package versions or do you have any idea what's causing my problem?

Thanks for your help!

Cheers,
Thomas

**Comment by aditya-malte** [comment]
Hello, 
Interesting to note.  Perhaps, as you said the environment might be a problem.Could you mail me at aditya.malte@gmail.com for further discussion.

Thanks and regards,
Aditya Malte

**Comment by mohammedayub44** [comment]
@aditya-malte What was the solution for this problem ? I get the same loss varaitions and predictions are nan's.

**Comment by saahil1292** [comment]
> Hi,
> I cloned the repo again, redid the preprocessing which seemed completely fine and started a new training job this morning. I didn't change anything at the model. I only added one cell for plotting the loss.
> My result with 200 epochs and your default values looks like this:
> 
> ![image](https://user-images.githubusercontent.com/34755160/55402421-be9b3500-5553-11e9-8e4a-fb7d04008a21.png)
> 
> I uploaded the two notebooks to this [repository](https://github.com/Puetz/csrnet-keras-issue/blob/master/Model.ipynb), so that you can have a look at them.
> My package version are
> 
> `Keras==2.2.4 tensorflow-gpu==1.13.1 scipy==1.2.1 numpy==1.16.2 Pillow==5.4.1 opencv-python==4.0.0.21`.
> 
> I tried using your versions but CUDA 9.0 did not work on my GPU. Do you think my bad results are related to the specific package versions or do you have any idea what's causing my problem?
> 
> Thanks for your help!
> 
> Cheers,
> Thomas

Hi Puetz, 
Did you find the solution to your problem?

I have tried training 'n' number of times but the model seems to overfit a lot. While Training MSE is coming around 0.5, test MSE goes highest to 3.5L.

Let me know if you get this solved. :)
